GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/GVFSLockTests.cs

```diff
@@ -102,15 +101,28 @@ private void OverwritingIndexShouldFail(string testFilePath)
             this.fileSystem.WriteAllText(testFilePath, testFileContents);
 
             this.Enlistment.WaitForBackgroundOperations().ShouldEqual(true, "Background operations failed to complete.");
-            MoveFileEx(
-                testFilePath, 
-                indexPath, 
-                (uint)(MoveFileFlags.MoveFileReplaceExisting | MoveFileFlags.MoveFileCopyAllowed)).ShouldBeFalse("GVFS should prevent renaming on top of index when GVFSLock is not held");
+
+            this.RenameAndOverwrite(testFilePath, indexPath).ShouldBeFalse("GVFS should prevent renaming on top of index when GVFSLock is not held");
             byte[] newIndexContents = File.ReadAllBytes(indexPath);
 
             indexContents.SequenceEqual(newIndexContents).ShouldBeTrue("Index contenst should not have changed");
 
             this.fileSystem.DeleteFile(testFilePath);
         }
+
+        private bool RenameAndOverwrite(string oldPath, string newPath)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))

```

Comment:
Every time I see a new platform check, I wonder if we're creating a maintenance headache for ourselves down the road. What are your thoughts on this? Is it a big enough issue to worry about yet? Is there a cleaner pattern that we could use to at least encapsulate the platform checks into one place? (I'm happy to keep deferring this if that's what makes sense, just wanted to bring it up for discussion)